system/mp/mpcomm.cpp

```diff
@@ -808,10 +808,9 @@ protected: friend class CMPPacketReader;
                                 EXCLOG(e, "MP: Failed to connect");
 #endif
                                 e->Release();
-                                if ((retrycount--==0)||(tm.timeout==MP_ASYNC_SEND))
+                                if ((--retrycount==0)||(tm.timeout==MP_ASYNC_SEND))
```

Comment:
Not sure that's the best way to fix - I would have added a +1 on the code that divides by retrycount